snuba/web/rpc/__init__.py

```diff
@@ -42,6 +42,14 @@ def metrics(self) -> MetricsWrapper:
             tags={"endpoint_name": self.__class__.__name__, "version": self.version()},
         )
 
+    @classmethod
+    def list_all_endpoint_names(cls) -> list[str]:
+        return [
+            name.split("__")[0]
+            for name in getattr(cls, "_registry").all_names()
```

Comment:
why do we use getattr rather than `cls._registry`?
is it bc python wont let you since RegisteredClass uses setattr for it?

Reply:
what if you did something like
```suggestion
            c.config_key().split("___")[0]
            for c in in cls.all_classes()
```
or add `all_names` as a function to `RegisteredClass` so you dont need to access `_registry`